src/tools/nix-dev-shell/shell.nix

```
@@ -1,18 +1,22 @@
-{ pkgs ? import <nixpkgs> {} }:
-let 
-  x = import ./x { inherit pkgs; };
+{pkgs ? import <nixpkgs> {}}: let
+  x = pkgs.callPackage ./x {};
 in
-pkgs.mkShell {
-  name = "rustc";
-  nativeBuildInputs = with pkgs; [
-    binutils cmake ninja pkg-config python3 git curl cacert patchelf nix
-  ];
-  buildInputs = with pkgs; [
-    openssl glibc.out glibc.static x
-  ];
-  # Avoid creating text files for ICEs.
-  RUSTC_ICE = "0";
-  # Provide `libstdc++.so.6` for the self-contained lld.
-  # Provide `libz.so.1`
-  LD_LIBRARY_PATH = "${with pkgs; lib.makeLibraryPath [stdenv.cc.cc.lib zlib]}";
-}
+  pkgs.mkShell {
+    strictDeps = true;
+    pname = "rustc-shell";
+    version = "none";
+
+    packages = let
+      p = pkgs;
+    in [
+      p.git
+      p.nix
+      p.glibc
+      x
+    ];
```

```suggestion
    packages = [
      pkgs.git
      pkgs.nix
      pkgs.glibc
      x
    ];
```
the alias just adds more noise